lib/scholar/cluster/k_means.ex

```diff
@@ -14,21 +14,28 @@ defmodule Scholar.Cluster.KMeans do
 
   ## Options
 
-    * `:num_clusters` - The number of clusters to form as well as the number of centroids to generate. Defaults to `8`.
+    * `:num_clusters` - The number of clusters to form as well as the number of centroids to generate. Required.
```

Comment:
Btw, why did you make them required? If 8 is a random number, perhaps a better default is 2? Or does Scipy make them required?

Reply:
I suggested making the number of clusters required mainly because since any choice we provide as default is arbitrary, I didn't see a reason for it being optional.

Reply:
I think the least arbitrary is 2 but I am fine with it being required or following Scipy.